File: xArm_Motion.py
#!/usr/bin/env python

import sys
import rospy
from geometry_msgs.msg import Pose

#API
from xarm.wrapper import XArmAPI

""" 
#######################################################
# Helper class for xArm interface with SDK 

# input: none
# output: none

# author: Mark Lee (MoonRobotics@cmu.edu)
# version: 1.0 (05/2023)
#######################################################
""" 

class xArm_Motion():
    def __init__(self, ip_addr):
        print(f" ---- creating xArm_Wrapper for ip {ip_addr}----")
        # self.stuff = 0
        self.ip = ip_addr

    def initialize_robot(self):
        print(" ---- initializing robot ----")
        self.arm = XArmAPI(self.ip)
        self.arm.motion_enable(enable=True)
        self.arm.set_mode(0)
        self.arm.set_state(state=0)
        self.arm.reset(wait=True)

    




    
if __name__ == "__main__":
    print(" ================ testing main of xArm_Motion wrapper ============ ")

File: xArm_motion.cpp
#include "ros/ros.h"
#include "std_msgs/Float64.h"
// #include "amiga_xarm/Patch.h"
#include "amiga_xarm_msgs/Patch.h"
#include "geometry_msgs/Point.h"
#include <xarm_api/xarm_driver.h>

#include <iostream>
#include <string>
#include <cmath>
#include <chrono>
#include <thread>

#include <signal.h>

#include <sensor_msgs/Joy.h>
#include <std_msgs/String.h>

""" 
#######################################################
# Helper class for xArm interface with SDK 

# input: none
# output: none

# author: Dominic Guri  (@andrew.cmu.edu)
# version: 1.0 (05/2023)
#######################################################
""" 

int main(int argc, char **argv)
{
  ros::init(argc, argv, "cornstalk_insertion");
  const std::string robot_ip = "192.168.1.213";
  ScrapperController scrapper(robot_ip, FREQUENCY);
  ros::spin();
}